src/test/java/org/alfresco/service/cmr/repository/datatype/DefaultTypeConverterTest.java

```diff
@@ -163,9 +169,12 @@ public void testFromString()
         assertEquals(Locale.FRANCE, DefaultTypeConverter.INSTANCE.convert(Locale.class, "fr_FR_"));
         
         assertEquals(new VersionNumber("1.2.3"), DefaultTypeConverter.INSTANCE.convert(VersionNumber.class, "1.2.3"));
-        
         assertEquals(new Period("period"), DefaultTypeConverter.INSTANCE.convert(Period.class, "period"));
         assertEquals(new Period("period|12"), DefaultTypeConverter.INSTANCE.convert(Period.class, "period|12"));
+        Map<String,String> periodMap = new HashMap<String, String>();
+        periodMap.put("periodType","month");
+        periodMap.put("expression","1");
+        assertEquals(new Period(periodMap), DefaultTypeConverter.INSTANCE.convert(Period.class, "month|1"));
```

Comment:
This isn't testing anything about the DefaultTypeConverter that isn't tested by the assertion above ("period|12"), if anything it's really testing is the `Period(Map)` constructor

Reply:
i.e. `DefaultTypeConverter.INSTANCE.convert(Period.class, periodMap)` is what should be tested here

Reply:
ok I change